sklearn/decomposition/tests/test_nmf.py

```diff
@@ -589,12 +644,75 @@ def test_nmf_custom_init_dtype_error():
     W = rng.random_sample((20, 15))
 
     with pytest.raises(TypeError, match="should have the same dtype as X"):
-        NMF(init='custom').fit(X, H=H, W=W)
+        estimator(init='custom').fit(X, H=H, W=W)
 
     with pytest.raises(TypeError, match="should have the same dtype as X"):
         non_negative_factorization(X, H=H, update_H=False)
 
 
+def test_nmf_close_minibatch_nmf():
+    # Test that the decomposition with standard and minbatch nmf
+    # gives close results
+    rng = np.random.mtrand.RandomState(42)
+    X = np.abs(rng.randn(48, 5))
+    nmf = NMF(5, solver='mu', init='nndsvdar', random_state=0,
+              max_iter=2000, beta_loss='kullback-leibler')
+    mbnmf = MiniBatchNMF(5, solver='mu', init='nndsvdar', random_state=0,
+                         max_iter=2000, beta_loss='kullback-leibler',
+                         batch_size=48)
+    W = nmf.fit_transform(X)
+    mbW = mbnmf.fit_transform(X)
+    assert_array_almost_equal(W, mbW, decimal=2)
```

Comment:
`decimal=2` is pretty low precision, I would hope that we can go at least up to `decimal=7` here, since you use `batch_size=n_samples`.

We should also test the case `batch_size<n_samples`, where the precision would probably need to be much lower.

Reply:
I have parametrized `batch_size`: the tests pass with `decimal=2` even when `batch_size<n_samples`... I'm not sure this was what you expected...

Reply:
Yes, the idea is to test that NMF and MiniBatchNMF give consistent results.
The previous test was only checking the routing, but the underlying implementations were the same, since it was using full batches.
With `batch_size < n_samples`, we also check that the batch implementation works.